### PR TITLE
feat: slicing navbar UI

### PR DESCRIPTION
**Pull request by azhargraha**
### Changes
- slicing navbar UI
- slicing sidebar UI
- remove `Vue.vscode-typescript-vue-plugin` extension recommendation

### Preview
- **transparent variant**
_navbar_
image
_sidebar_
image

- **minimal variant**
_navbar_
image
_sidebar_
image

### Evidence
title: feat slicing navbar UI
project: J-Site
participants: @azhargraha @Ibwedagama @agunghide @doohanas

### CODE REVIEW

File: assets/icons/common/chevron-down.svg
Comment (agunghide) [inline assets/icons/common/chevron-down.svg:1]: Ask:
I am wondering, why insert an image if you had use library `NuxtIcon` that can use the icons available on iconify?

ref: 
https://nuxt.com/modules/icon#iconify-dataset

File: components/Navbar.vue [hunk 1]
```diff
@@ -0,0 +1,235 @@
+<template>
+  <nav
+    :class="[
+      'fixed top-0 flex w-full items-center gap-4',
+      props.size,
+      props.variant,
+      props.navMenuPosition,
+    ]"
+  >
```
[the remark this answers is not in the document]
Comment (Ibwedagama) [reply components/Navbar.vue:9]: IMO we should structure them like this:

```
<header>
  <nav>
    <a><img src="mylogo" /></a>
    <ul>
      <li><a>Link</a></li>
      <li><a>Link</a></li>
      <li><a>Link</a></li>
    <ul>
  </nav>
 </header>
```

wdyt @azhargraha ? 🤔

[the remark this answers is not in the document]
Comment (azhargraha) [reply components/Navbar.vue]: noted, i've changed it [here](https://github.com/jabardigitalservice/j-site-public/pull/45/commits/9214930467687cba5b8fd17b1e3c4c6819caa733)

File: assets/icons/common/chevron-down.svg
Comment (Ibwedagama) [reply assets/icons/common/chevron-down.svg:1]: this is a great question, but you're referring to another module

we use this module https://nuxt.com/modules/icons

you're right, for common icons like an arrow, chevron, close, and trash we could use Iconify. They are providing a lot of icons out of the box. 

But in my experience, the UI designers pick icons from anywhere they want, sometimes they just want to create their own icons. 

It is hard to know where a specific icon comes from. So it's easier to export them as SVG, and render them as SVG

that's why we use this module
https://nuxt.com/modules/icons

**Review by doohanas — APPROVED** [review]

File: assets/icons/common/chevron-down.svg
Comment (agunghide) [reply assets/icons/common/chevron-down.svg:1]: Ah i see, so this case would be better for custom icon.
But the best way is make a deal with UI designer about what type icon they use for the design 🤣.
Thanks for the insight @Ibwedagama @azhargraha

**Review by agunghide — APPROVED** [review]

**Review by Ibwedagama — APPROVED** [review]

**Pull request merged**